scripts/clean_obsolete_strings.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+'''
+This script is used to remove obsolete strings from all locales,
+and reformat files accordingly to the source content (en-US).
+'''
+
+from compare_locales.parser import getParser
+from compare_locales.serializer import serialize
+import os
+import subprocess
+import sys
+
+
+def extractFileList(repository_path):
+    '''
+        Extract the list of supported files. Store the relative path and ignore
+        specific folders for products we don't want to modify (e.g. SeaMonkey).
+    '''
+
+    supported_formats = [
+        '.dtd',
+        '.properties',
+        '.ftl',
+    ]
+
+    excluded_folders = (
+        'calendar',
+        'chat',
+        'editor',
+        'extensions',
+        'mail',
+        'other-licenses',
+        'suite',
+    )
+
+    file_list = []
+    for root, dirs, files in os.walk(repository_path, followlinks=True):
+        for filename in files:
+            filename = os.path.relpath(
+                os.path.join(root, filename),
+                repository_path
+            )
+            # Ignore specific products
+            if filename.startswith(excluded_folders):
```

Comment:
The better way to do this is to manipulate the `dirs` list in-place, if `root` is `repository_path`.

Reply:
Also, you're iterating through `.hg`, AFAICT?

Reply:
Yes, it currently iterates over `.hg`. I'll just add it to the excluded folders, instead of ignoring all hidden folders. I hope I got the first one correctly :)